simtools/corsika/primary_particle.py

```diff
@@ -14,28 +14,33 @@ class PrimaryParticle:
 
     Parameters
     ----------
-    corsika7_id : int
-        CORSIKA7 ID of the primary particle.
-    name : str
-        Common name of the primary particle.
-    pdg_id : int
-        PDG ID of the primary particle.
+    particle_id_type : str
+        Type of the primary particle ID ('corsika7_id', 'common_name', or 'pdg_id').
+    particle_id : int or str
+        The actual ID of the primary particle.
     """
 
-    def __init__(self, corsika7_id=None, name=None, pdg_id=None):
+    def __init__(self, particle_id_type=None, particle_id=None):
         self._logger = logging.getLogger(__name__)
 
         self._corsika7_id = None
         self._name = None
         self._pdg_id = None
         self._pdg_name = None
 
-        if corsika7_id is not None:
-            self.corsika7_id = corsika7_id
-        if name is not None:
-            self.name = name
-        if pdg_id is not None:
-            self.pdg_id = pdg_id
+        if bool(particle_id_type) != bool(particle_id):
```

Comment:
Why not just make both arguments required?

Reply:
We are using `PrimaryParticle()` without any arguments in the code (although at least for the corsika configuration we need to have this value set).

Reply:
I'd probably would prefer using `None` instead of an instance of particle that contains `None`s to represent "no particle configured".

Reply:
Yes, I understand that motivation. However, as it is used right now we rely on that a PrimaryParticle object is defined. So changing this year, would mean to introduce if statements to check for None. Would prefer for now to leave it as it is to allow to have a 'default' PrimaryParticle object.